x-pack/osquerybeat/ext/osquery-extension/pkg/encoding/encoding.go

```diff
@@ -145,6 +146,17 @@ func convertValueToString(fieldValue reflect.Value, flag EncodingFlag) (string,
 		}
 		return strconv.FormatFloat(val, 'f', -1, 64), nil
 
+	case reflect.Struct:
+		switch fieldValue.Type() {
+		case reflect.TypeOf(time.Time{}):
+			val := fieldValue.Interface().(time.Time).Unix()
+			if !flag.has(EncodingFlagUseNumbersZeroValues) && val == 0 {
+				return "", nil
+			}
+			return strconv.FormatInt(val, 10), nil
+		}
+		return "", fmt.Errorf("unsupported struct type: %s", fieldValue.Type())
+
```

Comment:
for time.TIme particularly I think it could be nice to accept some tag parameters eg: 
```go
... `osquery:"timestamp,format:unix"`
... `osquery:"timestamp,format:rfc3339,tz:utc"`
```

since I can see fields which we might want to use not always just unix epoch